src/web/app/pages-instructor/instructor-course-edit-page/instructor-course-edit-page.component.html

```diff
@@ -22,18 +22,32 @@
     </span>
   </div>
   <div class="card-body fill-plain">
-    <div>
+    <form #courseForm="ngForm" (ngSubmit)="onSaveCourse()">
       <div class="form-row form-group">
         <label class="col-sm-3">Course ID:</label>
         <div class="col-sm-3">
-          <input class="col-xs-12 form-control" maxlength="40" tabindex="1" [(ngModel)]="course.courseId" disabled>
+          <input class="col-xs-12 form-control" type="text" tabindex="1"
+                 #courseId="ngModel" [(ngModel)]="course.courseId" name="courseId"
+                 [maxlength]="FormValidator.COURSE_ID_MAX_LENGTH" required disabled>
+          <div [hidden]="courseId.valid || (courseId.pristine && courseId.untouched)"
+               class="invalid-field">
+            <i class="fa fa-exclamation-circle" aria-hidden="true"></i>
+            The field Course ID should not be empty.
+          </div>
         </div>
       </div>
       <div class="form-row form-group">
         <label class="col-sm-3">Course Name:</label>
         <div class="col-sm-9">
-          <input class="form-control" ngbTooltip="The name of the course, e.g. Software Engineering."
-                 maxlength="64" tabindex="2" [(ngModel)]="course.courseName" [disabled]="!isEditingCourse">
+          <input class="form-control" ngbTooltip="Enter the name of the course, e.g. Software Engineering."
+                 type="text" tabindex="2" [disabled]="!isEditingCourse" name="courseName"
+                 [(ngModel)]="course.courseName" #courseName="ngModel"
+                 [maxlength]="FormValidator.COURSE_NAME_MAX_LENGTH" required>
+          <div [hidden]="courseName.valid || (courseName.pristine && courseName.untouched)"
+               class="invalid-field">
+            <i class="fa fa-exclamation-circle" aria-hidden="true"></i>
+            The field Course Name should not be empty.
```

Comment:
There is a method called `displayEmptyFieldMessage` in `instructor-course-student-edit-page.component.ts`: 
```typescript
/**
 * Displays message to user stating that the field is empty.
 */
displayEmptyFieldMessage(fieldName: string): string {
  return `The field '${fieldName}' should not be empty.`;
}
```
Although it is okay to leave it there, maybe for better consistancy we should consider refactoring this method out so that the course edit page and other future frontend validations can reuse these constants.

Reply:
Yes I copied the method at first (which is why the wording is the same) but then figured it's out of the scope of this PR. I feel like it would be better addressed separately as a new issue. I was going to get around to it eventually but you can go ahead and open it :)